release_scripts/Packages.Microsoft/README.md

```diff
@@ -0,0 +1,89 @@
+
+> Publishing to the Microsoft Repo may only be completed on corp-net. Support for non-corp-net access is not available at the time of this writing. Repo CLI docs may be accessed [here](http://csd-linux-publishing-service.azurewebsites.net/).
+
+## Requirements
+
+### Install Docker Desktop
+You can install Docker Desktop for macOS and Windows [here](https://www.docker.com/products/docker-desktop).
+
+### Generate Required Files for Repo CLI
+[Repo CLI](http://csd-linux-publishing-service.azurewebsites.net/client#commands) is the required API to publish Linux packages to Packages.Microsoft. Two files are required to use this tool:
+1. **A certificate** used for authenticating with the tool.
+2. **A JSON config file** as listed on the [Repo CLI docs](http://csd-linux-publishing-service.azurewebsites.net/client#commands).
+
+#### Download Certificate
+The certificate is hosted on the team's key vault account in Azure, under 'Certificates'. Download the certificate as a PEM file and include it in this directory (Packages.Microsoft). Rename the file as **private.pem**.
+
+#### Download Config
+> This section needs to be completed.
+
+## Build Docker Container
+Complete the below command below to create a Docker container with dependencies installed for publishing.
+
+From the root repo directory, run:
+```sh
+docker build -t mssqlcli-publish-msftrepo -f release_scripts/Packages.Microsoft/Dockerfile . --no-cache
+```
+
+## Run Docker Container
+This will run the Docker container built using the previous command.
+```sh
+docker run -it mssqlcli-publish-msftrepo bash
+```
+
+## Install Repo CLI
+In your Docker container, complete the command below to install the Repo CLI, the tool used to publish packages to Packages.Microsoft.
+
+```sh
+apt-get -y install azure-repoapi-client
+```
+
+## Publish to Packages.Microsoft Using Repo CLI
+The `publish.sh` script will upload deb and rpm packages to Packages.Microsoft by consulting the `supported_repos_prod.json` or `supported_repos_testing.json` file. This data is used to find supported repositories located in Packages.Microsoft.
+
+`publish.sh` takes `'testing'` or `'prod'` parameters as its second argument. Uploading to the testing distribution is effective for testing the installation and usage of mssql-cli on various Linux systems without affecting production.
+
+### Publish to Testing
+From the mssql-cli folder root, make the following command to upload packages to all repositories in the `supported_repos_testing.json` file:
+```sh
+release_scripts/Packages.Microsoft/publish.sh $(pwd) 'testing'
+```
+
+Visit the [testing](#testing-distribution-downloads) section for more information on downloading builds from the testing distribution.
+
+### Publish to Production
+When ready for production, change `'testing'` with `'prod'`, which will upload to all repositories in the `supported_repos_prod.json` file:
+```sh
+release_scripts/Packages.Microsoft/publish.sh $(pwd) 'prod'
+```
+
```

Comment:
We should add a note about what todo if there is an error or need to refresh the binaries with a new version. Can we just re-run the tool with the same package versions?

Reply:
Do you mean package versions of mssql-cli? Yes.

Reply:
I'm not sure what I mean. I just remember you mentioned there was an issue with stdio and the repro tool. If there is an issue, can you refresh my memory?

Reply:
ops, wrong thread. Cool, we should doc this since many repo's (nuget, pypi) don't support this (or didn't in the recent past)

Reply:
Let me run this back with the Repo Client team to confirm.